### PR TITLE
Feature/advanced ai racer

### PR DESCRIPTION
**Pull request by MatthewWhite**
A lot of this work is from @drmacintosh 

This adds an AI racer who navigates the track using nodes defined in XML.  It's still pretty basic, but the AI racer handles the same as a human driver, and can swerve when taking sharp turns.  There are some issues still to be worked out with the pathfinding to make it more robust, but this will work for a first iteration.

### CODE REVIEW

File: CS483/CS483/Kartaclysm/Components/ComponentAIDriver.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	{
+	}
+
+	HeatStroke::Component* ComponentAIDriver::CreateComponent(
+		HeatStroke::GameObject* p_pGameObject,
+		tinyxml2::XMLNode* p_pBaseNode,
+		tinyxml2::XMLNode* p_pOverrideNode)
+	{
+		std::string strTargetGUID("");
+		HeatStroke::GameObject* pTargetGameObject = nullptr;
```
Comment (bradcooper314) [inline CS483/CS483/Kartaclysm/Components/ComponentAIDriver.cpp:40]: I'm assuming these are remnants of future changes? They don't seem to do anything...

File: CS483/CS483/Kartaclysm/Components/ComponentKartController.h [hunk 2]
```diff
@@ -93,6 +95,7 @@ namespace Kartaclysm
 		//--------------------------------------------------------------------------
 		HeatStroke::GameObject* m_pGameObject;
 		int m_iPlayerNum;
+		bool m_bAI;
```
Comment (bradcooper314) [inline CS483/CS483/Kartaclysm/Components/ComponentKartController.h:98]: Should add an IsAI() method for future use

**Review by bradcooper314 — APPROVED** [review]
Very hacked together with lots of missing features, but core pathing seems to be mostly okay. Shift rift is a little tricky for the AI on the bottom turns. Next priority (this sprint or next) should be to have Racing State account for AI racers so race doesn't end whenever first racer crosses finish line (unintentional).

**Review by ghost — COMMENTED** [review]
good shit

**Pull request merged**